### DISCUSSION TITLE
EF6 support

### DISCUSSION THREAD

**Discussion by statler** [LuckyPennySoftware/AutoMapper#4268] [General]
I can find nowhere that states Automapper does not support EF6. The docs for v11+ literally say state **AutoMapper now targets .Net Standard 2.1 and doesn’t work on .Net Framework**. That is a reference to the framework, which is a completely different thing. **EF6 is .Net Standard 2.1 compatible** and I am running it in .Net 7, _not framework_. I could not find it anywhere that EF6 is not supported, so must conclude that this has not in fact been communicated, if it is actually correct.

That said, the changes I have proposed are literally all that is required to make it work, so it is very easy to continue supporting a very large part of the automapper universe that may still be stranded on v10 without this change. There are projects  have to remain on EF6 due to its compatibility with specifie database types that EFCore doesnt target.

**Comment by lbargaoanu** [comment]
Then it's very easy to do that in your own fork.

**Reply by statler** [reply]
Fair enough. Pretty sure I am not the only one with projects stuck on EF6. Regardless, if this is the position, then can you please clarify the documentation as it is not clear on this and it would be best if others didn't unnecessarily spend time solving the same issue unnecessarily. Also, if you can leave this discussion here, I will just leave a breadcrumb to the solution for anyone who needs it below.

https://github.com/statler/AutoMapper
https://www.nuget.org/packages/AutoMapperEF6.Compat/

**Reply by jbogard** [reply]
I think what we can do in the docs is make it clear that EFCore is the only query provider we target, test for, and support. EF6 is special in that we used to support it, but at this point it's in the same bucket of "maybe it works" as all the other query providers out there. Which there are many.

I dunno if we could make hooks in the API to make it easier to support other query providers.
